plugin-server/src/cdp/services/legacy-onevent-compare.service.ts

```
@@ -0,0 +1,250 @@
+        // These are plugins that use the legacy storage
+        inputs['legacy_plugin_config_id'] = { value: pluginConfig.id }
+    }
+
+    for (const [key, value] of Object.entries(pluginConfig.config)) {
+        inputs[key] = { value }
+    }
+
+    hogFunction.inputs = inputs
+
+    return hogFunction
+}
+
+export class LegacyOneventCompareService {
+    legacyPluginExecutorService: LegacyPluginExecutorService
+    hogFunctionsByPluginConfigId: Record<string, HogFunctionType | null>
+
+    constructor(private hub: Hub) {
+        this.legacyPluginExecutorService = new LegacyPluginExecutorService(this.hub)
+        this.hogFunctionsByPluginConfigId = {}
+    }
+
+    get shouldCompare(): boolean {
+        return this.hub.DESTINATION_MIGRATION_DIFFING_ENABLED === true && this.hub.TASKS_PER_WORKER === 1
+    }
+
+    getOrCreateHogFunction(pluginConfig: PluginConfig): HogFunctionType | null {
+        // We do essentially what the python migration code does to convert a plugin config into a hog function
+        if (!this.hogFunctionsByPluginConfigId[pluginConfig.id]) {
+            let hogFunction: HogFunctionType | null = null
+            try {
+                hogFunction = convertPluginConfigToHogFunction(pluginConfig)
+            } catch (e) {
+                status.error('Failed to convert plugin config to hog function', e)
+            }
+
+            this.hogFunctionsByPluginConfigId[pluginConfig.id] = hogFunction
+        }
+
+        return this.hogFunctionsByPluginConfigId[pluginConfig.id]
+    }
+
+    /**
+     * Temporary code path designed for 1-1 testing of inlined JS plugins with legacy plugins.
+     */
+    async runOnEvent(
+        pluginConfig: PluginConfig,
+        onEvent: PluginMethodsConcrete['onEvent'],
+        event: PostIngestionEvent,
+        onEventPayload: ProcessedPluginEvent
+    ): Promise<void> {
+        if (!this.shouldCompare) {
+            return onEvent(onEventPayload)
```

if we put the below onEvent into a try catch we should do the same here, right?

```
 try {
            // Execute the operation
            await onEvent(onEventPayload)
        }
```

---

We only do it below to track the comparison. If we are early exiting here then we can just let it throw and the parent carries on with it